akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -1114,34 +1118,43 @@ private[cluster] final class JoinSeedNodeProcess(seedNodes: immutable.IndexedSeq
  *
  * The supplied callback will be run, once, when current cluster member is `Up`.
  */
-private[cluster] class OnMemberUpListener(callback: Runnable) extends Actor with ActorLogging {
+private[cluster] class OnMemberStatusChangedListener(callback: Runnable, prefer: MemberStatus) extends Actor with ActorLogging {
```

Comment:
The parameter name "prefer" is misleading. It does not simply prefer, it actually listens to that event only.